File: src/main/java/fr/lewon/web/bot/entities/enums/WinnerType.java
package fr.lewon.web.bot.entities.enums;

public enum WinnerType {
	
	HERO("hero"),
	CHAMPION("champion");
	
	private final String strValue;
	
	private WinnerType(String strValue) {
		this.strValue = strValue;
	}

	public String getStrValue() {
		return strValue;
	}
	
	public static WinnerType findByStrValue(String strValue) {
		for (WinnerType wt : values()) {
			if (wt.strValue.equals(strValue)) {
				return wt;
			}
		}
		return null;
	}
	
}

File: src/main/kotlin/fr/lewon/bot/hh/entities/enums/WinnerType.kt
package fr.lewon.bot.hh.entities.enums

enum class WinnerType(val strValue: String) {
    HERO("hero"), CHAMPION("champion");

    companion object {
        fun findByStrValue(strValue: String): WinnerType? {
            for (wt in values()) {
                if (wt.strValue == strValue) {
                    return wt
                }
            }
            return null
        }
    }

}